engine/api/application.go

```diff
@@ -272,19 +286,19 @@ func (api *API) deleteApplicationHandler() service.Handler {
 		projectKey := vars[permProjectKey]
 		applicationName := vars["applicationName"]
 
-		proj, errP := project.Load(api.mustDB(), projectKey)
-		if errP != nil {
-			return sdk.WrapError(errP, "deleteApplicationHandler> Cannot laod project")
+		proj, err := project.Load(api.mustDB(), projectKey)
```

Comment:
I think it's a bit sad to load the project to get its ID, whereas the projectKey is an identifier.
we should consider `application.LoadByProjectKeyAndName`